src/main/scala/org/constellation/trust/TrustManager.scala

```diff
@@ -0,0 +1,72 @@
+package org.constellation.trust
+
+import cats.effect.Concurrent
+import cats.implicits._
+import org.constellation.consensus.StoredSnapshot
+import org.constellation.domain.schema.Id
+import org.constellation.domain.trust.TrustData
+import org.constellation.primitives.concurrency.SingleRef
+
+object TrustManager {
+
+  def calculateReputation(snapshots: List[StoredSnapshot]): Map[Id, Double] =
+    snapshots
+      .flatMap(_.checkpointCache.flatMap(_.checkpointBlock.flatMap(_.observations)))
+      .groupBy(o => o.signedObservationData.data.id)
+      .mapValues(_.size.toDouble) // TODO: wkoszycki add conversion List[Observation] -> Score
```

Comment:
@buckysballs how do I convert that ?

Reply:
This is good for the first implementation, we'll want to add logic for calculating a those Doubles that pertains to each ObservationEvent, but right not a count of Observations per Id will give us enough info to factor them into handleTrustScoreUpdate. For validator rewards, we will convert all Observations (and using associated logic) into `atb.interfaces.Experience` and then pass into `EigenTrustJ`, and then to `Rewards.validatorRewards` . But we can focus on `handleTrustScoreUpdate` and using it for snapshot selection for this pr.